botorch/models/approximate_gp.py

```diff
@@ -192,7 +192,7 @@ def __init__(
         Args:
             train_X: Training inputs (due to the ability of the SVGP to sub-sample
                 this does not have to be all of the training inputs).
-            train_Y: Training targets (optional).
+            train_Y: Not used.
```

Comment:
interesting - why did this have the arg in the first place? Any reason (other than linting) not to get rid of this?

Reply:
Perhaps it had this so that `_SingleTaskVariationalGP` would have a similar signature to `SingleTaskVariationalGP`? In any case, I don't see a good reason to keep it, especially since they differ in other ways at this point.

Reply:
I think we should keep it to support inducing point allocators that take into account the target values. To this end, we'd need to add `outputs` (in addition to `inputs`) to the signature of
```
inducing_point_allocator.allocate_inducing_points
```
This would be ignored for the default variance reduction strategy, but would be necessary for e.g. @henrymoss's [Inducing Point Allocation for Sparse Gaussian Processes in High-Throughput
Bayesian Optimisation](https://proceedings.mlr.press/v206/moss23a/moss23a.pdf). This would also require supporting `outputs` in the quality functions, but not a lot else as far as I can tell.